internal/operator-controller/controllers/clusterextensionrevision_controller.go

```diff
@@ -149,6 +149,11 @@ func (c *ClusterExtensionRevisionReconciler) reconcile(ctx context.Context, rev
 
 	rres, err := c.RevisionEngine.Reconcile(ctx, *revision, opts...)
 	if err != nil {
+		if rres != nil {
+			l.Error(err, "revision reconcile failed", "report", rres.String())
+		} else {
+			l.Error(err, "revision reconcile failed")
+		}
```

Comment:
I don't know if it's much of an improvement, but if we want to eliminate the duplication here we could do something like

```suggestion
        var kvs []string
		if rres != nil {
		    kvs = append(kvs, "report", rres.String())
		}
		l.Error(err, "revision reconcile failed", kvs...)
```
or something like that